### DISCUSSION TITLE
Formik Props update

### DISCUSSION THREAD

**Discussion by tusharpari** [jaredpalmer/formik#3729] [Q&A]
My form component look like below -

```
<Formik
        validationSchema={validationSchema}
        initialValues={initialValues)
        onSubmit={onSubmit}
        enableReinitialize
      >
        {(formProps) => ( 
            <Component formProps={formProps} />
         )}
    </Formik>  
```   

        
 Let say, If we want to update formik initialValues/values inside **Component**
 Can we update the initialValues/values like below.. I mean is it a good practice ?
1.  Object.assign(formProps.initialValues,  ...{ //some data  })
2. Object.assign(formProps.values, { ... formProps.initialValues } )

Thanks

**Comment by tusharpari** [comment]
@jaredpalmer 
Any thoughts on above query ?

Thanks

**Comment by karfau** [comment]
From what I understood you should use `setValues` (or `setFieldValue`?) from `formProps`.
